iodata/test/test_factorial2.py

```diff
@@ -0,0 +1,43 @@
+import unittest
+import numpy as np
+from .. import overlap
+
+factorial2 = overlap.factorial2
+
+
+class TestFactorial2(unittest.TestCase):
```

Comment:
In IOData, we don't use the `unittest` module, nor test classes, nor `self.assert...` methods. Instead, we use Pytest, which is designed to be fully functional with simple test functions. It makes writing and extending tests a lot easier. One of the nice features of Pytest is that it can deduce informative error messages from normal `assert` statements.

For testing numerical data, I'd recommend `pytest.approx`. See:

- https://stackoverflow.com/questions/8560131/pytest-assert-almost-equal
- https://docs.pytest.org/en/latest/reference/reference.html#pytest.approx

Our current code is using `numpy.testing` instead of `pytest.approx`, because a lot of that code predates `pytest.approx`. We should migrate eventually for the sake of robustness. Feel free to use `pytest.approx` already.